File: Domain/Progress.cs
namespace Domain
{
    public class Progress
    {
        public int Id { get; set; }
        public Decimal Value { get; set; }
        public string Description { get; set; }
        public DateTime Date { get; set; }
        public Goal Goal { get; set; }
        public Category Category { get; set; }
    }
}

File: goals-progress/src/app/models/Progress.ts
import dayjs from "dayjs";
import { Category } from "./Category";

export interface Progress{
    id: number;
    value: number;
    date: dayjs.Dayjs | Date;
    description?: string;
    category?: Category;
    categoryId?: number;
}

export interface ProgressFormValues {
    id: number;
    value: number | null;
    date: dayjs.Dayjs | Date;
    description?: string;
    category?: Category;
    categoryId?: number;
}